File: src/main/java/io/vertigo/shiny/components/media/photo/ShinyPhoto.java
package io.vertigo.shiny.components.media.photo;

import io.vertigo.core.lang.Assertion;
import io.vertigo.shiny.components.ShinyComponent;

public record ShinyPhoto(
		String title,
		String url,
		String alt) implements ShinyComponent {

	@Override
	public String type() {
		return "ShinyPhoto";
	}

	public ShinyPhoto {
		Assertion.check()
				.isNotBlank(url, "URL cannot be blank");
	}
}

File: banshee/src/models/ShinyPhoto.ts
import { ShinyComponent } from '../ShinyComponent';

export class ShinyPhoto implements ShinyComponent {
  title?: string;
  url: string;
  alt?: string;
  type: string = 'photo';

  constructor(title: string, url: string, alt: string) {
    this.title = title;
    this.url = url;
    this.alt = alt;
  }
}
